src/app/core/entity/entity-mapper.service.ts

```diff
@@ -141,6 +141,29 @@ export class EntityMapperService {
     return result;
   }
 
+  /**
+   * Saves an array of entities that are possibly heterogeneous, i.e.
+   * the entity-type of all the entities does not have to be the same.
+   * This method should be chosen whenever a bigger number of entities needs to be
+   * saved
+   * @param entities The entities to save
+   */
+  public async saveAll(entities: Entity[]): Promise<any> {
+    const rawData = entities.map((e) =>
+      this.entitySchemaService.transformEntityToDatabaseFormat(e)
+    );
+    entities.forEach((entity) => {
+      this.sendUpdate(entity, entity._rev === undefined ? "new" : "update");
+    });
+    const results = await this._db.putAll(rawData);
+    results
+      .filter((res) => res?.ok)
+      .map((res, idx) => {
+        entities[idx]._rev = res.rev;
```

Comment:
Let's move the `sendUpdate` call here so only entities that have been correctly saved/created are emitted.